features/google/src/model/importers/utils/lang-constants.js

```diff
@@ -13,8 +13,19 @@ export const langConstants = {
     },
 };
 export function matchRegex(path, importer) {
+    // There can be non standard spaces in the path. Replace them with a normal space.
+    const sanitizedPath = path.replace(
+        /[\u00a0\u1680\u2000-\u200b\u202f\u205f\u3000\ufeff]/,
+        " "
+    );
     const importerName = importer.constructor.name;
-    const normalizedPath = path.normalize("NFC");
+    const normalizedPath = sanitizedPath.normalize("NFC");
+    const normalizedRegex = RegExp(
+        langConstants[importerName].regex
+            .toString()
+            .normalize("NFC")
```

Comment:
Are you sure we need to normalise the expressions themselves in this way? Seems odd... We define the expressions, it's not external input, so we should be able to keep them in a well formed manner, no?

Reply:
How will we make sure that the special characters in the regex have the same bytes as their equivalent characters in the string?

Reply:
If you introduce a literal by the IDE, it will be already normalized.

Reply:
With NFC? It is not the case with VSC

Reply:
Well, it should be normalized. I assumed it was normalized in its "C" form (NFC) might be one of the others.